File: sorting_algorithm.jl
function bubbleSort(arr)
    n = length(arr)
    for i in 1:n-1
        for j in 1:n-i
            if arr[j] > arr[j+1]
                arr[j], arr[j+1] = arr[j+1], arr[j]
            end
        end
    end
    return arr
end

arr = [64, 34, 25, 12, 22, 11, 90]
println("Sorted array: $(bubbleSort(arr))")

File: SortingAlgorithm.hs
module SortingAlgorithm where

bubbleSort :: Ord a => [a] -> [a]
bubbleSort [] = []
bubbleSort [x] = [x]
bubbleSort xs = bubbleSort' xs (length xs)

bubbleSort' :: Ord a => [a] -> Int -> [a]
bubbleSort' xs 0 = xs
bubbleSort' xs n = bubbleSort' (bubble xs) (n - 1)

bubble :: Ord a => [a] -> [a]
bubble [x] = [x]
bubble (x:y:zs)
    | x > y     = y : bubble (x : zs)
    | otherwise = x : bubble (y : zs)

main :: IO ()
main = do
    let arr = [64, 34, 25, 12, 22, 11, 90]
    putStrLn $ "Sorted array: " ++ show (bubbleSort arr)
